ote/src/cljs/ote/app/controller/transit_visualization.cljs

```diff
@@ -408,99 +384,163 @@
-                  (do
-                    (comm/get! (str "transit-visualization/" service-id "/route-differences")
-                               {:params {:date1 (time/format-date-iso-8601 earlier-date)
-                                         :date2 (time/format-date-iso-8601 later-date)
-                                         :route-hash-id (ensure-route-hash-id route)}
-
-                                :on-success (tuck/send-async! ->RouteDifferencesResponse)})
-                    (-> app
-                        (assoc-in [:transit-visualization :route-differences-loading?] true)
-                        (assoc-in [:transit-visualization :compare :date1]
-                                  earlier-date)
-                        (assoc-in [:transit-visualization :compare :date2]
-                                  later-date)
-                        (assoc-in [:transit-visualization]
-                                  (fetch-trip-data-for-dates (:transit-visualization app)
-                                                             service-id
-                                                             route
-                                                             earlier-date
-                                                             later-date)))))))
-
-(define-event LoadServiceChangesForDateResponse [response detection-date]
   {}
-  (let [date-filter (if (= "now" (:scope app))
+  (let [service-id (get-in app [:params :service-id])
+        date1 (get-in app [:transit-visualization :compare :date1])
+        date2 (get-in app [:transit-visualization :compare :date2])
+        route (get-in app [:transit-visualization :selected-route])
+        goog-date1 (goog.date.DateTime. date1)
+        goog-date (goog.date.DateTime. date)
+        date-after-date1? (t/after? goog-date goog-date1)
+        earlier-date (if date-after-date1? date1 date)
+        later-date (if date-after-date1? date date1)]
+    (cond
+      (or (and date1 date2) (t/equal? goog-date1 goog-date)) ;; Re-selection of day pair after to replace previous day pair selection
+      (-> app
+          (assoc-in [:transit-visualization :compare :date1] date)
+          (update-in [:transit-visualization :compare] remove-date2-keys)
+          (assoc-in [:transit-visualization :route-dates-selected-from-calendar?] true))
+
+      (nil? date2)
+      (do
+        (comm/get! (str "transit-visualization/" service-id "/route-differences")
+                   {:params {:date1 (time/format-date-iso-8601 earlier-date)
+                             :date2 (time/format-date-iso-8601 later-date)
+                             :route-hash-id (ensure-route-hash-id route)}
+
+                    :on-success (tuck/send-async! ->RouteDifferencesResponse)})
+        (-> app
+            (assoc-in [:transit-visualization :route-differences-loading?] true)
+            (assoc-in [:transit-visualization :compare :date1]
+                      earlier-date)
```

Comment:
(assoc-in [:transit-visualization :compare :date1]
                      earlier-date)

(assoc-in [:transit-visualization :compare :date1] earlier-date)

Reply:
Done